lib/awesome_print.rb

```diff
@@ -1,41 +1,26 @@
-# Copyright (c) 2010-2016 Michael Dvorkin and contributors
-#
-# Awesome Print is freely distributable under the terms of MIT license.
-# See LICENSE file or http://www.opensource.org/licenses/mit-license.php
-#------------------------------------------------------------------------------
-#
-# AwesomePrint might be loaded implicitly through ~/.irbrc or ~/.pryrc
-# so do nothing for subsequent requires.
-#
-unless defined?(AwesomePrint::Inspector)
-  %w(awesome_method_array string method object class kernel).each do |file|
-    require "awesome_print/core_ext/#{file}"
-  end
+require 'awesome_print/formatters'
+require 'awesome_print/inspector'
+require 'awesome_print/formatter'
+require 'awesome_print/version'
+
+module AwesomePrint
+  class << self
+    attr_accessor :defaults, :force_colors
 
-  require 'awesome_print/custom_defaults'
-  require 'awesome_print/inspector'
-  require 'awesome_print/formatter'
-  require 'awesome_print/version'
-  require 'awesome_print/core_ext/logger' if defined?(Logger)
-  #
-  # Load the following under normal circumstances as well as in Rails
-  # console when required from ~/.irbrc or ~/.pryrc.
-  #
-  require 'awesome_print/ext/active_record'  if defined?(ActiveRecord)  || AwesomePrint.rails_console?
-  require 'awesome_print/ext/active_support' if defined?(ActiveSupport) || AwesomePrint.rails_console?
-  #
-  # Load remaining extensions.
-  #
-  if defined?(ActiveSupport.on_load)
-    ActiveSupport.on_load(:action_view) do
-      require 'awesome_print/ext/action_view'
+    # Class accessor to force colorized output (ex. forked subprocess where TERM
+    # might be dumb).
+    #---------------------------------------------------------------------------
```

Comment:
Probably way overkill for this PR, but may make sense to look into YARDoc later as a standard documentation framework.